src/gmail.js

```diff
@@ -1067,7 +1075,52 @@ var Gmail = function(localJQuery) {
         return result;
     };
 
+    api.tools.check_event_type = function(threadObj) {
+        const action_map = {
+            ""            : "add_to_tasks",
+            "^a"          : "archive",
+            "^k"          : "delete",
+            ""            : "delete_message_in_thread",
+            ""            : "delete_forever",
+            ""            : "delete_label",
+            ""            : "discard_draft",
+            ""            : "expand_categories",
+            ""            : "filter_messages_like_these",
+            ""            : "label",
+            ""            : "mark_as_important",
+            ""            : "mark_as_not_important",
+            ""            : "mark_as_not_spam",
+            ""            : "mark_as_spam",
+            ""            : "move_label",
+            ""            : "move_to_inbox",
+            ""            : "mute",
+            ""            : "read",
+            ""            : "save_draft",
+            ""            : "send_message",
+            ""            : "show_newly_arrived_message",
+            ""            : "star",
+            ""            : "undo_send",
+            ""            : "unmute",
+            ""            : "unread",
+            ""            : "unstar",
+            ""            : "new_email",
+            ""            : "poll",
+            ""            : "refresh",
+            ""            : "restore_message_in_thread",
+            ""            : "open_email",
+            ""            : "toggle_threads"
```

Comment:
I see what you're trying to do here, and it sort of makes sense. However if you *run* this code, it doesn't really work.

I suggest we comment out the unimplemented events for now, but keep them in as comments as a reminder of things which can/should be implemented later.

Does that sound OK?

Reply:
yes)